[ADD] sale_order_product_recommendation: SO product wizard

yajo opened the pull request:
This wizard that you can see now in your quotations will let you add products quickly based on the customer's recent purchase history.

~~Currently WIP due to a problem I can't seem to nail down: prices show always as zero:
![captura de pantalla de 2017-09-26 11-47-35](https://user-images.githubusercontent.com/973709/30853974-95ea480a-a2b0-11e7-8097-f3bcaf829650.png)~~

@Tecnativa

yajo committed:
[ADD] sale_order_product_recommendation: SO product wizard

This wizard that you can see now in your quotations will let you add products quickly based on the customer's recent purchase history.

sergio-teruel commented on sale_order_product_recommendation/wizards/sale_order_recommendation.py:
```diff
@@ -0,0 +1,142 @@
+        for one in self:
+            one.price_unit = one.product_id.with_context(
+                partner=one.partner_id.id,
+                pricelist_id=one.pricelist_id,
```
Change pricelist_id by pricelist
https://github.com/OCA/OCB/blob/9.0/addons/product/product.py#L789

yajo commented:
Thanks @sergio-teruel, it's working now! (I also had to add `.id` 😅)

rafaelbn commented:
Hi @Yajo , travis is red. Could you check it please?

yajo commented:
Last update includes products from current SO to wizard recommendations. It needed to:

- Autofill its included units
- Delete lines from SO if they were set to zero units
- Merge multiple lines if they share same product
- Update existing lines if qty has changed

Little feature, big job. It should be working in any case.

yajo commented:
OK so this is 💚 now finally

pedrobaeza committed:
[FIX] sale_order_product_recommendation: Limit after populating results

We can't limit results on read_group, because we can't set the same
criteria to order results, and thus the N groups populated might be
not the top ones looked for.

pedrobaeza reviewed (approved):
After I have fixed one thing

carlosdauden commented on sale_order_product_recommendation/wizards/sale_order_recommendation.py:
```diff
@@ -0,0 +1,177 @@
+            [
+                ("order_partner_id", "child_of",
+                 self.order_id.partner_id.commercial_partner_id.id),
+                ("order_id.date_order", ">=", start),
```
order_id.date_order can be replace with create_date (if little diferences are assumed) or create related field and set index=True to better performance

carlosdauden replied:
Execute Sql sentence with OVER clause can be another option to performance

carlosdauden committed:
[IMP] sale_order_product_recommendation: Improve performance

yajo commented:
Can we merge then?

pedrobaeza commented:
Please validate latest change by @carlosdauden

rafaelbn reviewed (approved):
Tested functionally in runbot :+1:

rafaelbn commented:
Shouldn't we migrate this module to v10 or v11 @Yajo ?

yajo commented:
I guess

pedrobaeza commented:
v11 as per our roadmap, but not yet.
